robusta_krr/core/integrations/prometheus/metrics/base_metric.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import abc
+import asyncio
+import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Callable, TypeVar
+
+from robusta_krr.core.abstract.strategies import ResourceHistoryData
+from robusta_krr.core.models.config import Config
+from robusta_krr.core.models.objects import K8sObjectData
+from robusta_krr.utils.configurable import Configurable
+
+if TYPE_CHECKING:
+    from ..loader import CustomPrometheusConnect
+
+REGISTERED_METRICS: dict[str, type[BaseMetricLoader]] = {}
+
+
+class BaseMetricLoader(Configurable, abc.ABC):
+    def __init__(self, config: Config, prometheus: CustomPrometheusConnect) -> None:
+        super().__init__(config)
+        self.prometheus = prometheus
+
+    @abc.abstractmethod
+    def get_query(self, namespace: str, pod: str, container: str) -> str:
+        ...
+
+    async def query_prometheus(
+        self, query: str, start_time: datetime.datetime, end_time: datetime.datetime, step: datetime.timedelta
+    ) -> list[dict]:
+        return await asyncio.to_thread(
```

Comment:
what happens when you get PrometheusNotFound?

Reply:
Here `PrometheusNotFound` would have been already raised. We have `self.prometheus` - an already working connection